bitcoin/src/blockdata/transaction.rs

```diff
@@ -1141,6 +1141,41 @@ internals::transparent_newtype! {
         pub fn assume_coinbase_ref(inner: &_) -> &Self;
     }
 }
+
+impl Coinbase {
+    /// Creates a `Coinbase` wrapper assuming this transaction is the first in a block.
+    ///
+    /// This method does not validate that the transaction is actually a coinbase transaction.
+    /// The caller must ensure that this transaction is indeed the first transaction in a valid block.
+    pub fn assume_first_transaction(tx: Transaction) -> Self {
+        Self(tx)
+    }
+
+    /// Creates a `Coinbase` reference assuming this transaction is the first in a block.
+    ///
+    /// This method does not validate that the transaction is actually a coinbase transaction.
+    /// The caller must ensure that this transaction is indeed the first transaction in a valid block.
+    pub fn assume_first_transaction_ref(tx: &Transaction) -> &Self {
+        Self::assume_coinbase_ref(tx)
+    }
+
+    /// Returns a reference to the inner transaction.
+    pub fn as_inner(&self) -> &Transaction { &self.0 }
+
+    /// Returns the inner transaction.
+    pub fn into_inner(self) -> Transaction { self.0 }
+
+    /// Computes the [`Txid`] of this coinbase transaction.
+    pub fn compute_txid(&self) -> Txid {
+        self.0.compute_txid()
+    }
+
+    /// Computes the [`Wtxid`] of this coinbase transaction.
+    pub fn compute_wtxid(&self) -> Wtxid {
```

Comment:
IIRC coinbase transactions have a constant `Wtxid` so this method looks misleading and probably shouldn't exist.